x-pack/plugins/ingest_manager/server/routes/epm/handlers.ts

```diff
@@ -156,15 +159,21 @@ export const installPackageHandler: RequestHandler<
     if (e instanceof IngestManagerError) {
       return defaultResult;
     }
-    // if there is an unknown server error, uninstall any package assets
+
+    // if there is an unknown server error, uninstall any package assets or reinstall the previous version if update
     try {
-      const installedPkg = await getInstallationObject({ savedObjectsClient, pkgName });
-      const isUpdate = installedPkg && installedPkg.attributes.version < pkgVersion ? true : false;
-      if (!isUpdate) {
+      if (!installedPkg) throw new Error('no installation exists');
+      if (installType === 'install' || installType === 'reinstall') {
```

Comment:
Can the type be install if there is an installed package? My understanding is the only way getInstallType returns install is if there isn't a package. And since we error if there is no package in the line above, we must have one here so it can't be install, right?

Reply:
this was only put here because if i don't check the installedPkg, i get a type error on line https://github.com/elastic/kibana/pull/76694/files/2ff6c9d7bcc8cf2d8f577639225d3198891428df#diff-a4f9e2e344a4b58d7209d56878afed76R171.

I changed it and used ts-ignore again